src/reducers/selectionReducer.js

```diff
@@ -35,6 +35,10 @@ const initialState = {
 };
 
 export default (state=initialState, { id: cardId, type }) => {
+  if(!cardId) {
```

Comment:
Ah, that's a good point.  Matching on the actions themselves seems like a much more robust approach; I'll make that change.